src/scanner.cc

```diff
@@ -199,7 +206,19 @@ class Scanner
         else if (lexer->lookahead == '\n')
         {
             advance(lexer);
-            lexer->result_symbol = PARAGRAPH_BREAK;
+
+            lexer->result_symbol = LINE_BREAK;
+
+            if (!lexer->lookahead || lexer->eof(lexer))
```

Comment:
technically you want them swapped since because `eof` is also `lexer->lookahead == 0`
also shouldn't we `mark_end`?

Reply:
You're right. I removed all of those excessive calls. I don't think `mark_end` is necessary here, mark_end only works like a save point to my understanding which we don't need.